checker.c

```diff
@@ -1,21 +1,48 @@
-#include <stdio.h>
-#include <assert.h>
+#include "checker.h"
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+float readtemperature(float temperature)
+{
+    if((temperature < TEMP_MIN) || (temperature > TEMP_MAX))
```

Comment:
Similar check is used for SOC too. Code duplication can be reduced here.

Reply:
Okay